mcp-voice-agent/agent.py

```
@@ -0,0 +1,270 @@
+                    except json.JSONDecodeError:
+                        return text
+                return response
+
+            # Build signature from schema
+            params = [
+                inspect.Parameter("context", inspect.Parameter.POSITIONAL_OR_KEYWORD, annotation=RunContext)
+            ]
+            ann = {"context": RunContext}
+
+            for name, ps in _props.items():
+                default = ps.get("default", inspect._empty if name in required else None)
```

Ruff (0.11.9)

Function definition does not bind loop variable `required`

(B023)